usaspending_api/etl/management/commands/load_transactions_in_delta.py

```diff
@@ -513,67 +514,77 @@ def update_awards(self):
-            )
 
+        # When handling date-parsing, require a separator for mmddYYYY, but not for YYYYmmdd, or there is no way to
+        # tell them apart from just regexp, and only YYYYmmdd has been seen without a separator.
+        # Each of these regexps allows for an optional timestamp portion, separated from the date by some character,
+        #   and the timestamp allows for an optional UTC offset.  In any case, the timestamp is ignored, though.
+        regexp_mmddYYYY = r"(\\d{2})(?<sep>[-/])(\\d{2})(\\k<sep>)(\\d{4})(.\\d{2}:\\d{2}:\\d{2}([+-]\\d{2}:\\d{2})?)?"
+        regexp_YYYYmmdd = r"(\\d{4})(?<sep>[-/]?)(\\d{2})(\\k<sep>)(\\d{2})(.\\d{2}:\\d{2}:\\d{2}([+-]\\d{2}:\\d{2})?)?"
+
+        def build_date_format_sql(
+            col: TransactionColumn, is_casted_to_date: bool = True, is_result_aliased: bool = True
+        ) -> str:
+            """Builder function to wrap a column in date-parsing logic.
+
+            It will either parse it in mmddYYYY format with - or / as a required separated, or in YYYYmmdd format
+            with or without either of - or / as a separtor.
+            Args:
+                is_casted_to_date (bool): if true, the parsed result will be cast to DATE to provide a DATE datatype,
+                    otherwise it remains a STRING in YYYY-mm-dd format
+                is_result_aliased (bool) if true, aliases the parsing result with the given ``col``'s ``dest_name``
+            """
+            mmddYYYY_fmt = f"""
+                (regexp_extract({bronze_table_name}.{col.source}, '{regexp_mmddYYYY}', 5)
+                || '-' ||
+                regexp_extract({bronze_table_name}.{col.source}, '{regexp_mmddYYYY}', 1)
+                || '-' ||
+                regexp_extract({bronze_table_name}.{col.source}, '{regexp_mmddYYYY}', 3))
+            """
+            YYYYmmdd_fmt = f"""
+                (regexp_extract({bronze_table_name}.{col.source}, '{regexp_YYYYmmdd}', 1)
+                || '-' ||
+                regexp_extract({bronze_table_name}.{col.source}, '{regexp_YYYYmmdd}', 3)
+                || '-' ||
+                regexp_extract({bronze_table_name}.{col.source}, '{regexp_YYYYmmdd}', 5))
+            """
+            if is_casted_to_date:
+                mmddYYYY_fmt = f"""CAST({mmddYYYY_fmt}
+                            AS DATE)
+                """
+                YYYYmmdd_fmt = f"""CAST({YYYYmmdd_fmt}
+                            AS DATE)
+                """
+            sql_snippet = f"""
+                CASE WHEN regexp({bronze_table_name}.{col.source}, '{regexp_mmddYYYY}')
+                          THEN {mmddYYYY_fmt}
+                     ELSE {YYYYmmdd_fmt}
+                END{' AS ' + col.dest_name if is_result_aliased else ''}
+            """
+            return sql_snippet
+
+        def handle_column(col, bronze_table_name, is_result_aliased=True):
             if col.handling == "cast":
```

Comment:
If you changed the call to `handle_column` below to `build_date_format_sql`, you don't need to add the `is_result_aliased` argument to this function here, which is never used in its original called location and doesn't really make sense to be allowed for the literal case.